signingscript/sign.py

```diff
@@ -707,3 +715,84 @@ def _owner_filter(tarinfo_obj):
 
     log.info("autograph wrote signed_file %s to %s", from_, to)
     return to
+
+
+async def sign_hash_with_autograph(context, from_, fmt, to=None):
+    """Signs a hash with autograph, injects it into the file, and writes the result to arg `to` or `from_` if `to` is None.
+
+    Args:
+        context (Context): the signing context
+        from_ (str): the source file to sign
+        fmt (str): the format to sign with
+        to (str, optional): the target path to sign to. If None, overwrite
+            `from_`. Defaults to None.
+
+    Raises:
+        Requests.RequestException: on failure
+
+    Returns:
+        str: the path to the signed file
+
+    """
+    if fmt != _AUTOGRAPH_SIGN_HASH_FMT:
+        raise SigningScriptError(
+            "No autograph servers found supporting hash signing of format {}".format(fmt)
+        )
+    cert_type = task.task_cert_type(context)
+    servers = get_suitable_signing_servers(context.signing_servers, cert_type, [fmt])
+    if not servers:
```

Comment:
Not a blocker. We seem to repeat this pattern all over. I'm almost thinking we could put in a `raise_on_empty_list=True` to `get_suitable_signing_servers`.